### DISCUSSION TITLE
Payment Records in contract1

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx-documentation#44] [Q&A] [answered]
I recently upgraded to the latest version of opencrx (v 5.2). I have been going through the contract1 model. I discovered a new class "PaymentRecord". I will like to know more about the use cases this class intends to solve.

**Comment by nyemiks** [comment]
In addition to the above question I will to know the meaning of this fields

pain1UltmtDbtrNm

pain1UltmtDbtrPstlAdrStrtNm

pain1UltmtCdtrNm

pain1UltmtCdtrPstlAdrStrtNm

pain1UltmtCdtrCtryOfRes

pain1CdtrPstlAdrStrtNm

pain1CdtrAgtFinInstnId

pain1RmtInfStrdCdtrRefInfTp

pain1RmtInfUstrd

pain1CdtrAcctCcy


in other words, I will like to know the meaning of the abbreviated words
like Ultmt, Rmt, Inf, Dbtr, Adr, Strt, Nm, Cdtr, Pstl. This will allow me to know how to use the class.

**Comment by christoph-mueller-crixp** [comment]
the class PaymentRecord relates to ISO 20022 - you should find answers to all your questions at https://www.iso20022.org/

**Reply by nyemiks** [reply]
Okay. I have gone through the links. The link you provided answers my second question. As regards the first question. What is the purpose of having a payment record under a contract ? Let's say the underlying contract is an invoice.

**Comment by christoph-mueller-crixp** [comment, marked as answer]
payments are typically based on some kind of "exchange" (e.g. goods or services in exchange for payment), and the terms of such an "exchange" are typically captured with a contract - hence it seems quite natural to associate payments with contracts

**Reply by nyemiks** [reply]
This clarifies.
